### PR TITLE
[IMP][8.0] account_payment_return: Add Bank charge and change payment…

### PR DESCRIPTION
**Pull request by carlosdauden**
… mode in returned invoices

### CODE REVIEW

File: account_payment_return/models/account_invoice.py [hunk 1]
```diff
@@ -31,3 +31,14 @@ class AccountInvoice(models.Model):
         string='Payment returned',
         help='Invoice has been included on a payment that has been returned '
              'later.')
+    bank_charge = fields.Float(
```
Comment (pedrobaeza) [inline account_payment_return/models/account_invoice.py]: It should be called "bank_fee" (but you can put oldname="bank_charge" to keep your compatibility)

File: account_payment_return/models/account_invoice.py [hunk 2]
```diff
@@ -31,3 +31,14 @@ class AccountInvoice(models.Model):
...
+        string='Bank charge',
+        compute='_compute_bank_charge',
+        help='Bank charge', readonly=True,
+        digits=dp.get_precision('Account'))
+
+    def _compute_bank_charge(self):
+        if self.payment_returned and (self.residual > self.amount_total):
+            self.bank_charge = self.residual - self.amount_total
+        else:
+            self.bank_charge = 0.0
```
Comment (pedrobaeza) [inline account_payment_return/models/account_invoice.py]: This is not needed. It's the value by default

File: account_payment_return/models/payment_return.py [hunk 3]
```diff
@@ -76,6 +76,16 @@ class PaymentReturn(models.Model):
                    ('cancelled', 'Cancelled')],
         string='State', readonly=True, default='draft',
         track_visibility='onchange')
+    notes = fields.Text('Notes')
+    bank_charge = fields.Float(
```
Comment (pedrobaeza) [inline account_payment_return/models/payment_return.py]: bank_fee

**Comment by pedrobaeza** [comment]
You should put the bank fee in a separate account (for example, 626000 in spanish accounting).

One doubt I have: how do search using the invoice reference instead of the payment reference?

**Comment by carlosdauden** [comment]
The purpose of this module is not really manage bank fees in accounting terms, what you do is add these costs to the client's debt to the customer then you pay that debt. It is as if a "supplied" not affecting your income or expenses. Because otherwise should also generate additional customer invoice.

**Comment by pedrobaeza** [comment]
No, that's not true. You have to generate an accounting entry this way: 626000 against 430000 (or the receivable account). That can compensate the 626000 with 572000 entry that you generate when processing bank statements. On contrary, you will get an erroneus balance on 626000, because you have already charged this expense again to the customer.

**Comment by carlosdauden** [comment]
The search by number, reference, or some other data invoice is in our roadmap, if you can orient ...
The easy option is to add a new field related to invoices and use this field to select the payments, but there are already too many fields on that screen.
Another option is that the field is "compute" and overwrite the search ...
So far it has not been necessary because normally the return receipt is usually the first on the list since we ordered downward.

**Comment by rafaelbn** [comment]
Are you going to continue with this PR?

**Comment by pedrobaeza** [comment]
With new bank-payment method, this one is totally valid, so there's nothing more to do. Maybe a complement to bank-payment suite (additional module) to allow to exclude returned payments in the selection wizard.

**Pull request closed without merging**